src/Extensions/Hosting.Services/Internal/OmexHostedService.cs

```diff
@@ -0,0 +1,98 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Microsoft.Omex.Extensions.Abstractions;
+
+namespace Microsoft.Omex.Extensions.Hosting.Services
+{
+	/// <summary>
+	/// Wraps service run into HostedService
+	/// </summary>
+	internal sealed class OmexHostedService : IHostedService
+	{
+		private readonly IHostApplicationLifetime m_lifetime;
+		private readonly IOmexServiceRunner m_runner;
+		private readonly ILogger<OmexHostedService> m_logger;
+		private readonly CancellationTokenSource m_tokenSource;
+
+		public OmexHostedService(IOmexServiceRunner runner, IHostApplicationLifetime lifetime, ILogger<OmexHostedService> logger)
+		{
+			m_runner = runner;
+			m_lifetime = lifetime;
+			m_logger = logger;
+			m_tokenSource = new CancellationTokenSource();
```

Comment:
Alternatively, you can merge this with the definition above

Reply:
I like to have all the initialization in one place so I would rather leave it here